src/Evolution/Systems/Cce/Initialize/ZeroNonSmooth.hpp

```diff
@@ -0,0 +1,99 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <memory>
+#include <string>
+
+#include "DataStructures/SpinWeighted.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Evolution/Systems/Cce/Initialize/InitializeJ.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class ComplexDataVector;
+/// \endcond
+
+namespace Cce {
+namespace InitializeJ {
+
+/*!
+ * \brief Initialize \f$J\f$ on the first hypersurface to be vanishing, finding
+ * the appropriate angular coordinates to be continuous with the provided
+ * worldtube boundary data.
+ *
+ * \details Internally, this performs an iterative solve for the angular
+ * coordinates necessary to give rise to a vanishing gauge-transformed J on the
+ * worldtube boundary. The parameters for the iterative procedure are determined
+ * by options `ZeroNonSmooth::AngularCoordinateTolerance` and
+ * `ZeroNonSmooth::MaxIterations`. The resulting `J` will necessarily
+ * have vanishing first radial derivative, and so will typically not be smooth
+ * (only continuous) with the provided Cauchy data at the worldtube boundary.
+ */
+struct ZeroNonSmooth : InitializeJ {
+  struct AngularCoordinateTolerance {
+    using type = double;
+    static std::string name() noexcept { return "AngularCoordTolerance"; }
+    static constexpr OptionString help = {
+        "Tolerance of initial angular coordinates for CCE"};
+    static type lower_bound() noexcept { return 1.0e-14; }
+    static type upper_bound() noexcept { return 1.0e-3; }
+    static type default_value() noexcept { return 1.0e-10; }
+  };
+
+  struct MaxIterations {
+    using type = size_t;
+    static constexpr OptionString help = {
+        "Number of linearized inversion iterations."};
+    static type lower_bound() noexcept { return 10; }
+    static type upper_bound() noexcept { return 1000; }
+    static type default_value() noexcept { return 300; }
+  };
+
+  struct ErrorIfNotConverged {
```

Comment:
Overall this is fine, but I don't love bools with `Not BLAH` in them... Maybe `RequireConvergence`?